x-pack/legacy/plugins/apm/typings/elasticsearch.ts

```diff
@@ -6,6 +6,12 @@
 
 import { StringMap, IndexAsString } from './common';
 
+export interface BoolQuery {
+  must_not: Array<Record<string, any>>;
```

Comment:
Why do we need this type?

Reply:
It's only used in `transaction_groups/fetcher.ts` so I can move it there if you prefer? Just thought it might be useful elsewhere so opted for adding it here.

Reply:
Another way of writing this would be:

```ts
type BoolQuery =  Record<"must_not" | "should" | "filter", Array<Record<string, any>>>;
```

Not sure what I prefer.

Reply:
It's more that I'm wondering why we need it at all?

Reply:
It's not possible to push to an empty array in an object without a type:
https://www.typescriptlang.org/play/#code/MYewdgzgLgBCBGArGBeGBvAUDHMCGATgQFwwDaAupgL4DcmCiAdIQUwA4CuEAFgBTo8pAOQ8ApgE9h1AJRA

Reply:
Ahh yeah, another argument to use inline ternary/spreads 😛

Reply:
Hehe, I'd still prefer this for readability. Plus, this way the dev gets intellisense, and warnings if they try to add invalid properties eg `must_filter` or something silly :)